### PR TITLE
Homework5: Eldar Dniaprouski

### PR DESCRIPTION
**Pull request by eldar-dniaprouski**

### CODE REVIEW

File: src/home_work_05/task_02_decorators.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import all_func
+from functools import wraps
+
+
+def result_decorator(some_func):
+    @wraps(some_func)
+    def wrapper(*args, **kwargs):
+        res = some_func(*args, **kwargs)
+        with open('results.txt', 'a') as res_file:
+            res_file.write('Function name: {}\n'.format(some_func.__name__))
```
Comment (belovmd) [inline src/home_work_05/task_02_decorators.py:17]: also we can add date time :)

File: src/home_work_05/task_03_get_ranges.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+get_ranges([4, 7, 10]) // "4,7,10"
+get_ranges([2, 3, 8, 9]) // "2-3,8-9"
+"""
+
+
+def get_ranges(int_list):
+    range_list = []
+    sub_list = []
+    for elem in int_list:
+        if elem - 1 not in int_list or elem + 1 not in int_list:
```
Comment (belovmd) [inline src/home_work_05/task_03_get_ranges.py]: Parentheses wouldn't be redundant here ))

Comment (belovmd) [inline src/home_work_05/task_03_get_ranges.py]: O(n**2) it can be solved as O(n)

File: src/home_work_05/task_04_imdb.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    make_top250_file(source_string)
+    make_ratings_file(source_string)
+    make_years_file(source_string)
+    return
+
+
+def make_top250_file(source_data):
+    with open(os.path.join('data5', 'top250_movies.txt'), 'w+') as top250:
+        for line in source_data.split('\n'):
+            clear_line = line[31:line.find('(') - 1].lstrip(' ')
```
Comment (belovmd) [inline src/home_work_05/task_04_imdb.py]: magic numbers

File: src/home_work_05/task_04_imdb.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import os
+
+
+def imdb():
+    line_count, source_string = 1, ''
+    try:
+        with open(os.path.join('data5', 'ratings.list'), 'r') as source:
+            for line in source:
+                if 29 <= line_count < 279:
```
Comment (belovmd) [inline src/home_work_05/task_04_imdb.py]: magic numbers

File: src/home_work_05/ascii_histogram/ratings_histogram.txt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+...........................................|===|..|===|..|===|..|===|..|===|..|===|.
+...........................................|===|..|===|..|===|..|===|..|===|..|===|.
+...........................................|===|..|===|..|===|..|===|..|===|..|===|.
+....................................|===|..|===|..|===|..|===|..|===|..|===|..|===|.
+.............................|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|.
+.............................|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|.
+.............................|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|.
+...............|===|.........|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|.
+...............|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|.
+.|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|..|===|.
```
Comment (belovmd) [inline src/home_work_05/ascii_histogram/ratings_histogram.txt:67]: nice histogram :)

**Review by belovmd — CHANGES_REQUESTED** [review]
Please take a look at these minor issues

File: src/home_work_05/task_02_decorators.py (hunk 1 above)
Comment (eldar-dniaprouski) [reply src/home_work_05/task_02_decorators.py:17]: done

File: src/home_work_05/task_03_get_ranges.py (hunk 2 above)
Comment (eldar-dniaprouski) [reply src/home_work_05/task_03_get_ranges.py]: solved as O(n)

Comment (eldar-dniaprouski) [reply src/home_work_05/task_03_get_ranges.py]: solved as O(n) with simple if condition

File: src/home_work_05/task_04_imdb.py (hunk 3 above)
Comment (eldar-dniaprouski) [reply src/home_work_05/task_04_imdb.py]: removed magic numbers

File: src/home_work_05/task_04_imdb.py (hunk 4 above)
Comment (eldar-dniaprouski) [reply src/home_work_05/task_04_imdb.py]: removed magic numbers

**Review by belovmd — COMMENTED** [review]
9,5

**Pull request closed without merging**